master/internal/api_workspace.go

```diff
@@ -1317,6 +1318,94 @@ func getAutoGeneratedNamespaceName(ctx context.Context, wkspID int, tx *bun.Tx)
 	return &namespace, nil
 }
 
+func (a *apiServer) GetWorkspacesWithDefaultNamespaceBindings(ctx context.Context,
+	req *apiv1.GetWorkspacesWithDefaultNamespaceBindingsRequest,
+) (*apiv1.GetWorkspacesWithDefaultNamespaceBindingsResponse, error) {
+	k8ClusterNames := a.m.config.GetKubernetesClusterNames()
+	ns := db.Bun().NewSelect().
+		Table("workspace_namespace_bindings").
+		Column("workspace_id").
+		ColumnExpr("COUNT(*) as num_bindings").
+		Where("cluster_name IN (?)", bun.In(k8ClusterNames)).
+		GroupExpr("workspace_id")
+
+	var workspaceIds []int32
+	err := db.Bun().NewSelect().
+		With("ns", ns).
+		TableExpr("workspaces as w").
+		Column("w.id").
+		Join("LEFT JOIN ns on ns.workspace_id = w.id").
+		Where("ns.num_bindings < ? or ns.num_bindings IS NULL", len(k8ClusterNames)).
+		Scan(ctx, &workspaceIds)
+	if err != nil {
+		return nil, err
+	}
+
+	return &apiv1.GetWorkspacesWithDefaultNamespaceBindingsResponse{WorkspaceIds: workspaceIds}, nil
+}
+
+func (a *apiServer) BulkAutoCreateWorkspaceNamespaceBindings(ctx context.Context,
+	req *apiv1.BulkAutoCreateWorkspaceNamespaceBindingsRequest,
+) (*apiv1.BulkAutoCreateWorkspaceNamespaceBindingsResponse, error) {
+	k8ClusterNames := a.m.config.GetKubernetesClusterNames()
+	wsns := []model.WorkspaceNamespace{}
+	err := db.Bun().NewSelect().
+		Table("workspace_namespace_bindings").
+		Where("cluster_name IN (?)", bun.In(k8ClusterNames)).
+		Scan(ctx, &wsns)
+	if err != nil {
+		return nil, err
+	}
+	wkspBinding := map[int32][]string{}
+	for _, v := range wsns {
+		if _, ok := wkspBinding[int32(v.WorkspaceID)]; !ok {
+			wkspBinding[int32(v.WorkspaceID)] = []string{v.ClusterName}
+		} else {
+			wkspBinding[int32(v.WorkspaceID)] = append(wkspBinding[int32(v.WorkspaceID)], v.ClusterName)
+		}
+	}
+
+	if err = a.bulkAutoCreateWorkspaceNamespaceBindingsHelper(ctx, req.WorkspaceIds, wkspBinding); err != nil {
+		return nil, err
+	}
+	return &apiv1.BulkAutoCreateWorkspaceNamespaceBindingsResponse{}, nil
+}
+
+func (a *apiServer) bulkAutoCreateWorkspaceNamespaceBindingsHelper(
+	ctx context.Context, wkspIDList []int32, wkspBinding map[int32][]string,
+) error {
+	for _, wID := range wkspIDList {
+		if _, ok := wkspBinding[wID]; !ok {
+			_, err := a.SetWorkspaceNamespaceBindings(ctx,
+				&apiv1.SetWorkspaceNamespaceBindingsRequest{
+					WorkspaceId: wID,
+					ClusterNamespaceMeta: map[string]*workspacev1.WorkspaceNamespaceMeta{
+						"": {AutoCreateNamespaceAllClusters: true, ClusterName: ""},
+					},
+				})
+			if err != nil {
+				return err
+			}
+		} else if len(wkspBinding[wID]) != len(a.m.allRms) {
```

Comment:
nit: It might be worth having the be a `<` check instead, in case we get into a funky case where `len(wkspBinding[wID]) > len(a.m.allRms)`

Reply:
Do we have any way of preventing multiple namespace bindings?

Reply:
For a given workspace we should never have more than one binding per cluster. In some cases we might have stale bindings where we might have bindings related to a cluster that is no longer active.

Reply:
Do you think it is worth having a panic in the case where `len(wkspBinding[wID]) > len(a.m.allRms)` ?

Reply:
You made a good point about there being potential stale bindings from an cluster that's no longer active, so I don't know if that's worth adding a panic for, if we can still move forward while one of the clusters is down, we should be solid.

Reply:
I am filtering the wkspBinding map to only have clusters that are currently active (in the sql query from the prev function) so there technically should be no case where `len(wkspBinding[wID]) > len(a.m.allRms)`. So if we do have that something might be going very wrong

Reply:
If it's possible for a cluster to become in-active between the API calls, I'd still hold off on the panic; it would be a pretty rough situation to debug

Reply:
The cluster might be active in Kubernetes but to use it in our system again would require a master restart and config change
So I think a panic is appropriate?

Reply:
Didn't add the panic